executor/table_reader.go

```diff
@@ -228,13 +228,10 @@ func (e *TableReaderExecutor) buildResp(ctx context.Context, ranges []*ranger.Ra
 		SetKeepOrder(e.keepOrder).
 		SetStreaming(e.streaming).
 		SetFromSessionVars(e.ctx.GetSessionVars()).
+		SetFromInfoSchema(e.ctx.GetInfoSchema().(infoschema.InfoSchema)).
 		SetMemTracker(e.memTracker).
 		SetStoreType(e.storeType).
 		SetAllowBatchCop(e.batchCop)
-	// infoschema maybe null for tests
-	if is, ok := e.ctx.GetSessionVars().GetInfoSchema().(infoschema.InfoSchema); ok {
-		reqBuilder.SetFromInfoSchema(is)
-	}
```

Comment:
I think this part of logic is not the same as before? `nil` interface should be handled.

Reply:
I thought `GetInfoSchema` will always return something. I did not expect that `domain.InfoSchema()` will return nil.

Reply:
Fixed. Now request builder will check it by `is, ok := emptyInterce.(InfoSchema)`.